okhttp-android-support/src/main/java/okhttp3/internal/huc/JavaApiConverter.java

```diff
@@ -505,51 +504,30 @@ static String extractStatusLine(Map<String, List<String>> javaResponseHeaders)
   /**
    * Creates an OkHttp Response.Body containing the supplied information.
    */
-  private static ResponseBody createOkBody(final Headers okHeaders,
+  private static Body createOkBody(final Headers okHeaders,
       final CacheResponse cacheResponse) throws IOException {
     final BufferedSource body = Okio.buffer(Okio.source(cacheResponse.getBody()));
-    return new ResponseBody() {
-      @Override
-      public MediaType contentType() {
-        String contentTypeHeader = okHeaders.get("Content-Type");
-        return contentTypeHeader == null ? null : MediaType.parse(contentTypeHeader);
-      }
-
-      @Override
-      public long contentLength() {
-        return HttpHeaders.contentLength(okHeaders);
-      }
-
-      @Override public BufferedSource source() {
-        return body;
-      }
-    };
+    return new Body(okHeaders.get("Content-Type") != null
+        ? MediaType.parse(okHeaders.get("Content-Type"))
+        : null,
+        HttpHeaders.contentLength(okHeaders),
+        body);
   }
 
   /**
    * Creates an OkHttp Response.Body containing the supplied information.
    */
-  private static ResponseBody createOkBody(final URLConnection urlConnection) throws IOException {
+  private static Body createOkBody(final URLConnection urlConnection) throws IOException {
     if (!urlConnection.getDoInput()) {
       return null;
     }
 
     final BufferedSource body = Okio.buffer(Okio.source(urlConnection.getInputStream()));
-    return new ResponseBody() {
-      @Override public MediaType contentType() {
-        String contentTypeHeader = urlConnection.getContentType();
-        return contentTypeHeader == null ? null : MediaType.parse(contentTypeHeader);
-      }
-
-      @Override public long contentLength() {
-        String s = urlConnection.getHeaderField("Content-Length");
-        return stringToLong(s);
-      }
-
-      @Override public BufferedSource source() {
-        return body;
-      }
-    };
+    return new Body(urlConnection.getContentType() != null
+        ? MediaType.parse(urlConnection.getContentType())
```

Comment:
... if the media type is invalid, and nobody attempts to parse it, no big deal. This is relevant in particular for this: `text/plain; charset=utf-8; charset=ascii`